### PR TITLE
Use more types form the inscription schema 🧙🏼‍♂️

### PR DESCRIPTION
**Pull request by ivy-lli**
This change has the effect, that more types are generated out of the inscription schema from the core.

### CODE REVIEW

File: packages/protocol/src/data/part-data.ts [hunk 1]
```diff
@@ -1,78 +1,34 @@
...
+export type ConditionData = Pick<ConfigData, 'conditions'>;
 
-export type ConditionData = ElementAlternative;
+export type TaskData = Pick<ConfigData, 'persistOnStart' | 'task' | 'tasks'>;
 
-export interface TaskData {
-  task: WfTask;
-  tasks: WfTask[];
-  persistOnStart: boolean;
-}
```
Comment (ivy-lli) [inline packages/protocol/src/data/part-data.ts:43]: Do no longer define our own types, but extract the needed configs from the ConfigData.
This has the benefit, that e.g. the ts typechecker fails if the `persist` flag was renamed to `persistOnStart`

File: packages/protocol/src/defaults/common-defaults.ts [hunk 2]
```diff
@@ -131,6 +141,25 @@ export const DEFAULT_DATA: ElementData = {
...
+    throws: {} as ErrorDefinition,
+    link: '',
+    clientId: '',
+    operation: {} as SoapOperation,
+    properties: {},
+    exception: {} as SoapWsProcessException,
+    eventId: '',
+    method: 'GET',
+    response: {} as RestResponse,
+    target: {} as RestTarget
```
Comment (ivy-lli) [inline packages/protocol/src/defaults/common-defaults.ts:163]: this is the only downside of this change at the moment...
We need to tell TS about this default data which is currently not used. But with each finished element view we would need to do this anyway, so this is currently only a placeholder until it is replaced by the real defaults when we implement those parts.

File: packages/protocol/src/utils/type-helper.ts [hunk 3]
```diff
@@ -9,3 +9,9 @@ export type ValuesAsUnionDeep<T> = {
 declare const brand: unique symbol;
 
 export type Brand<T, TBrand> = T & { [brand]: TBrand };
+
+export type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends (k: infer I) => void ? I : never;
```
Comment (ivy-lli) [inline packages/protocol/src/utils/type-helper.ts:13]: This converts a type of unions to an intersected type, e.g.
`{name: string} | {description: string}` becomes to 
`{name: string} & {description: string}` or better 
`{name: string; description: string}`

File: packages/protocol/src/utils/type-helper.ts [hunk 4]
```diff
@@ -9,3 +9,9 @@ export type ValuesAsUnionDeep<T> = {
...
+
+export type Prettify<T> = {
+  [K in keyof T]: T[K];
+} & {};
```
Comment (ivy-lli) [inline packages/protocol/src/utils/type-helper.ts:17]: typehelper to see the real type, e.g. if you hover the ConfigData from above you only see something like: `type ConfigData = ElementTaskEndPage & ElementTaskSwitchGateway & ElementSubProcessCall & ElementDatabase & ... 27 more ... & ElementSignalStartEvent`
with `Prettify<ConfigData>` you see something like `type ConfigData = {
    page: string;
    output: ScriptMapCode;
    case: WfCase;
    tasks: WfTask[];
    call: ScriptMapCode;
    processCall: string;
    cache: Cache;
    query: DbQuery;
    ... 34 more ...;
    attachToBusinessCase: boolean;
}`

not used currently, but really helpful if you try to write such ts types 🧙

Comment (weissreto) [reply packages/protocol/src/utils/type-helper.ts:17]: Wenn ich das richtig verstehe funktioniert das so:

Element1Data {
a:string,
b:string}
Element2Data {
c:string,
d:string}

ConfigData = {
a:string
b:string
c:string
d:string}

Element1InscriptionConfig {
a:string}
Element2InscriptionConfig {
c:string}

Man nimmt also alle Typen vom Protocol mixt die zusammen damit man einen Typ mit allen Properties hat und man nacher wieder einzelne Typen für die Inscription Masken erstellen kann die jedoch einzelne Properties aus dem Ursprünglichen Typen nicht mehr drin haben?

Warum macht man nicht für jeden Type eine Mapping Funktion die Protocol Objekt auf Inscription Object mapped anstelle hier mit Type Magic zu arbeiten, die nur wirkliche TS Experten verstehen?

Vielleicht hab ich auch einfach etwas nicht richtig verstanden.

**Review by weissreto — COMMENTED** [review]
Siehe mein Kommentar

**Pull request merged**